utils/ENTSOE_capacity_update.py

```diff
@@ -6,78 +6,131 @@
 import pandas as pd
 import json
 import pathlib
+import requests
+import xmltodict
+import datetime
 
-from parsers.ENTSOE import ENTSOE_PARAMETER_DESC, ENTSOE_PARAMETER_GROUPS
+from parsers.ENTSOE import ENTSOE_PARAMETER_DESC, ENTSOE_PARAMETER_GROUPS, ENTSOE_DOMAIN_MAPPINGS
 
 ZONESFILE = pathlib.Path(__file__).parent.parent / "config" / "zones.json"
 
 
 def update_zone(zone, data, zonesfile):
-	with open(zonesfile) as zf:
-		zones = json.load(zf)
+        with open(zonesfile) as zf:
+                zones = json.load(zf)
 
-	if zone not in zones:
-		raise ValueError("Zone {} does not exist in the zonesfile".format(zone))
+        if zone not in zones:
+                raise ValueError("Zone {} does not exist in the zonesfile".format(zone))
 
-	zones[zone]["capacity"].update(data)
+        zones[zone]["capacity"].update(data)
 
-	with open(zonesfile, "w") as zf:
-		json.dump(zones, zf, indent=2)
-		zf.write("\n")
+        with open(zonesfile, "w") as zf:
+                json.dump(zones, zf, indent=2)
+                zf.write("\n")
 
 
 def aggregate_data(data):
-	"""Aggregates data the way it is stated in
-	   parsers.ENTSOE.ENTSOE_PARAMETER_GROUPS"""
-	
-	# choose the column with the most current data
-	# assume keys start with YYYY
-	sorted_keys = list(sorted(data.keys()))
-	data = data[sorted_keys[-1]]
+        """Aggregates data the way it is stated in
+       parsers.ENTSOE.ENTSOE_PARAMETER_GROUPS"""
+        categories = dict(ENTSOE_PARAMETER_GROUPS["production"])
+        categories.update(ENTSOE_PARAMETER_GROUPS["storage"])
 
-	categories = dict(ENTSOE_PARAMETER_GROUPS["production"])
-	categories.update(ENTSOE_PARAMETER_GROUPS["storage"])
+        aggregated = {}
+        for group, category_abbreviations in categories.items():
+                aggregated[group] = sum(data.get(c, 0) for c in category_abbreviations)
 
-	aggregated = {}
-	for group, category_abbreviations in categories.items():
-		category_descriptions = list(map(lambda x: ENTSOE_PARAMETER_DESC[x], category_abbreviations))
-		aggregated[group] = sum(data[c] for c in category_descriptions)
-
-	return aggregated
+        return aggregated
 
 
 def parse_args():
-	parser = argparse.ArgumentParser()
-	parser.add_argument("--zonesfile", default=ZONESFILE)
-	parser.add_argument("zone", help="The zone abbreviation (e.g. AT)")
-	parser.add_argument("data_file", help="The csv file from ENTSOE containing the installed capacities")
-	return parser.parse_args()
+        parser = argparse.ArgumentParser()
```

Comment:
It's everywhere in this file, 8 space indentation rather than the original 4.

Reply:
Indeed, I did this with a different computer - must have had a different setting there. I converted it to four spaces now.